src/Collector.cs

```diff
@@ -127,19 +142,30 @@ private void SendQueuedMessages(Socket socket)
         }
     }
 
-    private void Authenticate(Socket socket)
+    private bool Authenticate(Socket socket)
     {
-      var data = System.Text.Encoding.ASCII.GetBytes("hello version 1.0\n");
+      var data = System.Text.Encoding.ASCII.GetBytes("hello version dotnet/0.2.0\n");
       socket.Send(data);
-      data = System.Text.Encoding.ASCII.GetBytes(String.Format("authenticate {0}\n", _apiKey));
+      if(!ReceiveOk(socket)) return false;
+      data = System.Text.Encoding.ASCII.GetBytes($"authenticate {_apiKey}\n");
       socket.Send(data);
+      return ReceiveOk(socket);
+    }
+
+    private bool ReceiveOk(Socket socket)
+    {
+      byte[] buffer = new byte[3];
+      socket.Receive(buffer);
+      // I'm not including all of LINQ for this shit
+      // return InstrumentalOk.SequenceEqual(buffer);
+      return InstrumentalOk[0] == buffer[0] && InstrumentalOk[1] == buffer[1] && InstrumentalOk[2] == buffer[2];
```

Comment:
Negative, Ghost Rider.  [SequenceEqual](http://stackoverflow.com/questions/12342714/how-to-compare-two-arrays-of-bytes/12342719#12342719) is an extension method from LINQ, which I did not want to pull in for one equality check.  This will also be substantially faster, though that almost certainly does not matter.